src/nxt_file.c

```diff
@@ -312,7 +312,7 @@ nxt_file_rename(nxt_file_name_t *old_name, nxt_file_name_t *new_name)
  * Linux 2.6 does not use BKL.
  */
 
-#if (NXT_HAVE_FIONBIO)
+#if defined(NXT_HAVE_FIONBIO) && !defined(NXT_SOLARIS)
```

Comment:
```#if (NXT_HAVE_FIONBIO) && !(NXT_SOLARIS)```
All `NXT_*` macros should be cheked this way as we want to have an ability to unset them with `-DNXT_HAVE_FIONBIO=0`.